dds_web/web/user.py

```diff
@@ -122,19 +119,41 @@ def confirm_invite(token):
 )
 def register():
```

Comment:
```
def register():
    """Handles the creation of a new user"""
    form = dds_web.forms.RegistrationForm()

    # Two reasons are possible for the token to be None.
    # The most likely reason is that the session has expired (given by PERMANENT_SESSION_LIFETIME config variable)
    # Less likely is that the confirm_invite was not called before posting the registration form.
    if flask.session.get("invite_token") is None:
        flask.current_app.logger.info(
            "No token has been found in session when posting to register."
        )
        flask.flash(
            "Error in registration process, please go back and use the link in the invitation email again."
        )
        return flask.redirect(flask.url_for("auth_blueprint.index"))

    # Validate form - validators defined in form class
    if form.validate_on_submit():
        # Create new user row by loading form data into schema
        try:
            new_user = user_schemas.NewUserSchema().load(form.data)
        except Exception as err:
            # This should never happen since the form is validated
            # Any error catched here is likely a bug/issue
            flask.current_app.logger.warning(err)
            flask.flash("Error in registration process, please try again.")
            return flask.redirect(flask.url_for("auth_blueprint.index"))

        flask.flash("Registration successful!")
        return flask.make_response(flask.render_template("user/userexists.html"))

    # Go to registration form
    return flask.render_template("user/register.html", form=form)
```

Reply:
We don't need the token here neither disclose the claims and do extra handling of them.